### ISSUE TITLE
LOOT is failing after update

### ISSUE THREAD

**Issue by Astralogic** [loot/loot#1015]
Hi, below is my log with the errors I'm experiencing, I tried to uninstall/reinstall, I then uninstalled and used the 7z archive, still the problem remains, this only happened after updating to the latest version. I also tried running it as admin with no luck.

> [14:07:12.013215] [warning]: Chromium console message: Styling master document from stylesheets defined in HTML Imports is deprecated. Please refer to https://goo.gl/EGXzpw for possible migration paths.
> 
> [14:07:12.064632] [warning]: Blocking load of resource at https://fonts.googleapis.com/css?family=Roboto+Mono:400,700|Roboto:400,300,300italic,400italic,500,500italic,700,700italic
> 
> [14:07:13.001470] [warning]: No masterlist present at C:\Users\Calvin\AppData\Local\LOOT\Fallout4\masterlist.yaml
> 
> [14:07:20.523509] [error]: Exception while executing query: boost::filesystem::permissions: The system cannot find the path specified: "C:\Users\Calvin\AppData\Local\Temp\LOOT-Fallout4-e823-fa09-ed5a-2ea6\.git\objects\pack\pack-99c39f2960992e3e5ef9d6efb53c050591896648.pack"
> 
> [14:07:20.524572] [error]: Chromium console message from http://loot/ui/app.bundle.js at line 1: Error: Oh no, something went wrong! You can check your LOOTDebugLog.txt (you can get to it through the main menu) for more information.
>     at onFailure (http://loot/ui/app.bundle.js:1:462893)

Any help appreciated.
Thanks

**Comment by LostDragonist** [comment]
Does "C:\Users\Calvin\AppData\Local\Temp" exist on your PC?  Can you create a file in it and delete that file?

**Comment by Astralogic** [comment]
Yes, it exists and I can create files there.

**Comment by WrinklyNinja** [comment]
```
[14:07:20.523509] [error]: Exception while executing query: boost::filesystem::permissions: The system cannot find the path specified: "C:\Users\Calvin\AppData\Local\Temp\LOOT-Fallout4-e823-fa09-ed5a-2ea6.git\objects\pack\pack-99c39f2960992e3e5ef9d6efb53c050591896648.pack"
```

There's something strange going on because `C:\Users\Calvin\AppData\Local\Temp\LOOT-Fallout4-e823-fa09-ed5a-2ea6.git\objects\pack\pack-99c39f2960992e3e5ef9d6efb53c050591896648.pack` is found by recursive directory scanning and then permissions are set on it, so the only thing I can think of is that something else is removing the file between it being found and permissions being set.

Do you have any disk cleanup applications running at the same time? E.g. I think newer CCleaner versions have a service that cleans stuff up in the background.

**Comment by Astralogic** [comment]
I have nothing like that at all, nothing that deletes files, only avast, but it has never interfered with LOOT.

Chwecking the directory, the file is still there.

**Comment by WrinklyNinja** [comment]
Oh, so the file exists? If so, can you check if it's read-only, and try unchecking that box in the file's properties if it is?

Also, is the file at `C:\Users\Calvin\AppData\Local\Temp\LOOT-Fallout4-e823-fa09-ed5a-2ea6.git\objects\pack\pack-99c39f2960992e3e5ef9d6efb53c050591896648.pack` or `C:\Users\Calvin\AppData\Local\Temp\LOOT-Fallout4-e823-fa09-ed5a-2ea6\.git\objects\pack\pack-99c39f2960992e3e5ef9d6efb53c050591896648.pack`, i.e. is there a `.git` folder in `LOOT-Fallout4-e823-fa09-ed5a-2ea6` or a `LOOT-Fallout4-e823-fa09-ed5a-2ea6.git` folder? I was expecting the former but the error message shows the latter, that may be related.

**Comment by Astralogic** [comment]
The path is: C:\Users\Calvin\AppData\Local\Temp\LOOT-Fallout4-e823-fa09-ed5a-2ea6\.git\objects\pack\pack-99c39f2960992e3e5ef9d6efb53c050591896648.pack

The problem however just got stranger. The file was indeed set to read only, but changing it did nothing, every time I run LOOT it creates new files with new random names and tries to use those, the issue is 
 LOOT is creating the pack file as read only.

Changing it from read only is useless because it just makes new ones and sets them to read only.

Holy crap I broke it :(

**Comment by WrinklyNinja** [comment]
> The problem however just got stranger. The file was indeed set to read only, but changing it did nothing, every time I run LOOT it creates new files with new random names and tries to use those

That's intentional.

> the issue is LOOT is creating the pack file as read only.

That's not.

I just wanted to know if you were able to make it writable, doing that wasn't going to change anything.

**Comment by Astralogic** [comment]
Oh, well yes I can make them writable.

**Comment by Astralogic** [comment]
I'm going to try reverting back to the previous version of LOOT. Could someone tell me where I can find it please?

**Comment by CptMcSplody** [comment]
@Astralogic https://github.com/loot/loot/releases

**Comment by WrinklyNinja** [comment]
Turns out the pack file being read-only is normal git behaviour, so I'll need to do something about handling that better, and in doing so that may fix this issue. If I get anywhere I'll post test builds in here.

**Comment by Astralogic** [comment]
I tried rolling back but the same still occurs.

**Comment by WrinklyNinja** [comment]
Try with [this build](https://bintray.com/loot/snapshots/download_file?file_path=loot_0.13.3-34-g2ee6db0_update-api.7z). Note that it needs the [MSVC 2017 x86 redistributable](https://download.visualstudio.microsoft.com/download/pr/749aa419-f9e4-4578-a417-a43786af205e/d59197078cc425377be301faba7dd87a/vc_redist.x86.exe), so install that if you don't already have it.

**Comment by Astralogic** [comment]
Running that build unfortunately doesn't solve the problem, here is the latest log:

> [12:25:09.084169] [warning]: Blocking load of resource at https://fonts.googleapis.com/css?family=Roboto+Mono:400,700|Roboto:400,300,300italic,400italic,500,500italic,700,700italic

>[12:25:10.338166] [warning]: No masterlist present at C:\Users\Calvin\AppData\Local\LOOT\Fallout4\masterlist.yaml

>[12:25:39.303638] [error]: Exception while executing query: permissions: The system cannot find the path specified.
: "C:\Users\Calvin\AppData\Local\Temp\LOOT-Fallout4-28c2b19a-3197-4e15-87aa-f937570fd877\.git\objects\pack\pack-aaa68ee97a55fdbf080f372c88b9b1dcce6e184a.pack"

>[12:25:39.304420] [error]: Chromium console message from http://loot/ui/app.bundle.js at line 5008: Error: Oh no, something went wrong! You can check your LOOTDebugLog.txt (you can get to it through the main menu) for more information.
    at onFailure (http://loot/ui/app.bundle.js:5008:5818)

**Comment by WrinklyNinja** [comment]
OK, try replacing the `loot_api.dll` in that build with [this one](https://bintray.com/loot/snapshots/download_file?file_path=loot_api-0.13.7-36-g4635757_breaking-changes-win32.7z).

**Comment by Astralogic** [comment]
Excellent! It works now :D Thanks a lot wrinkly!

I guess the next release will have this solved?

**Comment by WrinklyNinja** [comment]
Fixed in bba6ea4920b10e67a28f3ef2ea350a5677342b6f.

**Comment by jerry1333** [comment]
So I'm getting similar issues with latest version (0.13.4) while previous was working fine.
And if I move content of d:\_DANE\TEMP\LOOT-Fallout4-177f-471b-dc23-15a5\ to c:\Users\Jarek\AppData\Local\LOOT\Fallout4\ LOOT is working fine saying "no masterlist update nessecary".

[LOOTDebugLog.txt](https://github.com/loot/loot/files/2461999/LOOTDebugLog.txt)

**Comment by WrinklyNinja** [comment]
😫 

I guess I could try copying the directory instead of renaming, then attempting to delete it afterwards and just continuing if that delete fails.

**Comment by jerry1333** [comment]
It's probably failing because I moved TEMP folder to other disk (saving SSD writes).
Anyway if You have debug version to test send a link :)

**Comment by WrinklyNinja** [comment]
If you've moved your TEMP folder just to save SSD writes, be aware that [testing has found](https://www.heise.de/newsticker/meldung/SSD-Langzeittest-beendet-Exitus-bei-9-1-Petabyte-3755009.html) that [you don't need to bother](https://techreport.com/review/27909/the-ssd-endurance-experiment-theyre-all-dead). It certainly was a problem with early SSDs, but the linked tests are from 2015 and 2016, so if anything things should be even better now.

As for my idea, I might get around to having a build for you to test later this week, but maybe next week.

**Comment by Astralogic** [comment]
You know what I'm glad you brought this back up because I have only just realized that the problem only seemed to be solved because since actually paying attention I have noticed that my load order never actually changes.

I will make sure this is actually happening and reply later with confirmation, but I have been rearanging the last few ESP's of my load order to see if I can get LOOT to change them in any way and so far nothing.

**Comment by WrinklyNinja** [comment]
That sounds like an unrelated issue, if it is an issue. LOOT won't make changes to your load order without reason, if you move stuff around it will try to preserve your current load order unless it contradicts what its algorithm comes up with, and it sounds like that's what's happening.

**Comment by Ortham** [comment]
@jerry1333 Please try with [this LOOT snapshot](https://bintray.com/loot/snapshots/download_file?file_path=loot_0.13.3-37-g75ec1dd_update-api.7z), but replace its `loot_api.dll` with [this one](https://bintray.com/loot/snapshots/download_file?file_path=loot_api-0.13.8-39-g3f4b617_fix-clone-error-win32.7z).

**Comment by gracjanski** [comment]
[20:11:49.052522] [warning]: Blocking load of resource at https://fonts.googleapis.com/css?family=Roboto+Mono:400,700|Roboto:400,300,300italic,400italic,500,500italic,700,700italic
[20:11:50.367598] [warning]: No masterlist present at C:\Users\Graci\AppData\Local\LOOT\FalloutNV\masterlist.yaml
[20:12:05.602469] [error]: Exception while executing query: boost::filesystem::rename: Access is denied: "E:\LOOT-FalloutNV-d9c2-589f-a941-6c71\.git", "C:\Users\Graci\AppData\Local\LOOT\FalloutNV\.git"

The link is not available...

**Comment by Ortham** [comment]
[This snapshot build](https://bintray.com/loot/snapshots/download_file?file_path=loot_0.13.4-26-g8335a77_update-api.7z) includes the updated DLL.

**Comment by gracjanski** [comment]
After changing the loot_api.dll I cant start loot:
![zwischenablage01](https://user-images.githubusercontent.com/44848523/48311106-c48b7c00-e59a-11e8-9852-44026739bcfb.jpg)

**Comment by Ortham** [comment]
That's because you replaced the LOOT API DLL with an incompatible version.

**Comment by gracjanski** [comment]
in new version still the same problem

**Comment by Ortham** [comment]
The fix isn't part of the latest release.

**Comment by Ortham** [comment]
Fixed in ecbe050ad699fbde5ab13f7614bd80feea2ee182.
